pandas/core/indexing.py

```diff
@@ -1683,10 +1687,6 @@ def _setitem_with_indexer_split_path(self, indexer, value):
                     "when setting with an iterable"
                 )
 
-            elif lplane_indexer == 0 and len(value) == len(self.obj.index):
```

Comment:
i think we need to retain the check for len(value) since we only want to catch boolean masking (though clearly we dont have tests with a mismatched len(value))

probably also need to check that pi is a listlike with matching length too.

Reply:
indexers.check_setitem_lengths may be useful here

Reply:
What would we want to do with empty indexers then?

Reply:
The reason I though this would be ok is, that empty indexers look here the same as all False boolean indexers. Value in is here empty too, so we have that mismatch

Reply:
i think `ser.loc[[]] = [1, 2, 3]` should raise kind of like `ser.loc[[1]] = [1, 2, 3, 4]` would

Reply:
Problem here is:

In case of ``[]`` we have the following parameters:

```
lplane_indexer=0
indexer = (array([], dtype=int64), array([1])
value = array([])
```

In case of ``[False]``  we get exactyl the same parametrization. We cant really decide here whether we got a boolean indexer or not.

Additionally,

```
df = DataFrame({"a": ["a"], "b": [1], "c": [1]})
df.loc[list([]), ["b"]] = 10 - df["c"]
df.loc[list([]), ["b"]] = 11
```

does not raise on master either. Was not introduced here

Reply:
That is a tough one.  Let's get another opinion.  @jreback: what do you expect to happen with `ser.loc[[]] = [1, 2, 3]`?

Reply:
ping @jreback ?

Reply:
coming back to this, yeah i think if we don't have equal len indexer we should always raise, so @jbrockmendel example should raise